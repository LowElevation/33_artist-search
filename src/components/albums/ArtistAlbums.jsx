import React from "react";
import PropTypes from "prop-types";

const ArtistAlbums = ({ title, id, date }) => {
  return (
    <figure>
      <figcaption>{title}</figcaption>
      <img src={`http://coverartarchive.org/release/${id}/front`} 
      />
      <div>{date}</div>
    </figure>
  
  );

};

ArtistAlbums.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  date: PropTypes.string,
};

export default ArtistAlbums;

// onError="this.src = 'https://fakealbumcovers.com/images/FakeAlbums/yungvinci_selfportrait.jpg'" 
