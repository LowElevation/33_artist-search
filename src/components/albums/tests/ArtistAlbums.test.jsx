import React from "react";
import { render, cleanup } from "@testing-library/react";
import ArtistAlbums from "../ArtistAlbums";

describe("ArtistAlbums component", () => {
  afterEach(() => cleanup());
  it("renders ArtistAlbums", () => {
    const { asFragment } = render(<ArtistAlbums artistName={""} albums={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
