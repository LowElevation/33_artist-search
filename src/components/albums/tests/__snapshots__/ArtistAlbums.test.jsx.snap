// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistAlbums component renders ArtistAlbums 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
